src/appleseed/foundation/image/colorspace.h

```diff
@@ -863,10 +884,36 @@ Color<T, 3> spectrum_to_ciexyz(
     return Color<T, 3>(x, y, z);
 }
 
+template <typename T, typename SpectrumType>
+Color<T, 3> spectrum_illuminance_to_ciexyz(const SpectrumType& spectrum)
+{
+    static_assert(
+        SpectrumType::Samples == 31,
+        "foundation::spectrum_illuminance_to_ciexyz() expects 31-channel spectra");
+
+    T x = T(0.0);
+    T y = T(0.0);
+    T z = T(0.0);
+
+    for (size_t w = 0; w < 31; ++w)
+    {
+        const T val = spectrum[w];
+        x += XYZCMFCIE19312Deg[0][w] * val;
```

Comment:
A reflectance spectrum is not directly converted to another color space. We convert a reflected spectrum that we get when the reflectance spectrum is illuminated by a certain illuminant.
LightingConditions specify what that illuminant would be. I don't think it makes much sense to use it here since an illuminance spectrum doesn't need any lighting condition.

Reply:
Maybe I should abstract XYZCMFCIE19312Deg into another class?